clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/deploy/ops/securitygroup/UpsertOpenstackSecurityGroupAtomicOperation.groovy

```diff
@@ -47,12 +57,38 @@ class UpsertOpenstackSecurityGroupAtomicOperation implements AtomicOperation<Voi
   * Task status:
   * curl -X GET -H "Accept: application/json" localhost:7002/task/1
   */
+
   @Override
   Void operate(List priorOutputs) {
     task.updateStatus BASE_PHASE, "Upserting security group ${description.name}..."
 
-    description.credentials.provider.upsertSecurityGroup(description.id, description.name, description.description, description.rules)
+    OpenstackClientProvider provider = description.credentials.provider
+
+    // Try getting existing security group, update if needed
+    SecGroupExtension securityGroup
+    if (StringUtils.isNotEmpty(description.id)) {
+      task.updateStatus BASE_PHASE, "Looking up existing security group with id ${description.id}"
+      securityGroup = provider.getSecurityGroup(description.id)
+      if (securityGroup == null) {
```

Comment:
Do we want to throw an operation exception on a missing security group?  I would think we would want to throw a 404 back so the client can probably display resource not found.  Feels like we need to have consistency here.